pkg/kube/kclient/client.go

```diff
@@ -115,14 +121,19 @@ func (n *informerClient[T]) AddEventHandler(h cache.ResourceEventHandler) {
 		Handler: h,
 	}
 	reg, _ := n.informer.AddEventHandler(fh)
+	n.handlerMu.Lock()
+	defer n.handlerMu.Unlock()
 	n.registeredHandlers = append(n.registeredHandlers, reg)
 }
 
 func (n *informerClient[T]) HasSynced() bool {
 	if !n.informer.HasSynced() {
 		return false
 	}
-	for _, g := range n.registeredHandlers {
+	n.handlerMu.RLock()
+	handlers := slices.Clone(n.registeredHandlers)
```

Comment:
seems no need to clone. HasSynced is just returning a flag

Reply:
The concern was the read on n.registeredHandlers.

Or you mean we can just lock across HasSynced calls since it is (supposed to be) fast?

Reply:
yes